crates/miden-tx/src/executor/exec_host.rs

```diff
@@ -210,7 +212,7 @@ where
             });
         }
 
-        Ok(Vec::new())
+        Ok(vec![AdviceMutation::extend_merkle_store(asset_witness.authenticated_nodes())])
```

Comment:
Question: here we assume that the `SmtLeaf` contains only a single key-value pair, right? If the leaves could contain multiple key-value pairs, then, I think, we'd also need to extend the advice map with the pre-image of the leaf hash.